alsoHeresTheInterviewQuestionIHad/alsoHeresTheInterviewQuestionIHad.js

```diff
@@ -0,0 +1,31 @@
+const fs = require("fs");
+const streetArray = fs.readFileSync("data.txt", "utf-8").split(",");
+let answer = "";
+let tallestBuilding = Math.max(...streetArray);
+let numberOfBuildings = streetArray.length;
+let tallestBuildingSoFar = 0;
+let bestArea = tallestBuilding * numberOfBuildings;
+let bannerOne = [tallestBuilding, numberOfBuildings];
+let bannerTwo = [tallestBuilding, numberOfBuildings];
+let bestBannerOne = [tallestBuilding, numberOfBuildings];
+let bestBannerTwo = [tallestBuilding, numberOfBuildings];
+for (let i = 0; i < numberOfBuildings - 1; i++){
+    if (streetArray[i] > tallestBuildingSoFar) {
+        tallestBuildingSoFar = streetArray[i];
+    }
+    bannerOne = [tallestBuildingSoFar, i + 1];
+    bannerTwo = [Math.max(...streetArray.slice(i + 1, streetArray.length)), numberOfBuildings - i - 1];

```

Comment:
This definitely works, but every time we shift the banner over, we have to recalculate the max height of the second banner. Is there a way to avoid this?

Reply:
It feels like this isn't a rhetorical question.

Reply:
;)

Reply:
Have I addressed everything except this issue?